packages/controller/main.js

```diff
@@ -900,27 +903,43 @@ function reportStatus() {
     //   elapsed: 6650000,     // ms since the start of the process
     //   timestamp: 864000000  // ms since epoch
     // }
-    pidUsage(process.pid, (err, stats) => {
-        // controller.s might be stopped, but this is still running
-        if (!err && states && states.setState && stats) {
-            states.setState(id + '.cpu', { ack: true, from: id, val: Math.round(100 * parseFloat(stats.cpu)) / 100 });
-            states.setState(id + '.cputime', { ack: true, from: id, val: stats.ctime / 1000 });
-            outputCount += 2;
-        }
-    });
+    try {
+        pidUsage(process.pid, (err, stats) => {
+            // controller.s might be stopped, but this is still running
+            if (!err && states && states.setState && stats) {
+                states.setState(id + '.cpu', {
+                    ack: true,
+                    from: id,
+                    val: Math.round(100 * parseFloat(stats.cpu)) / 100
+                });
+                states.setState(id + '.cputime', { ack: true, from: id, val: stats.ctime / 1000 });
+                outputCount += 2;
+            }
+        });
+    } catch (e) {
+        logger.error(`${hostLogPrefix} Cannot read pidUsage data : ${e.message}`);
+    }
 
-    const mem = process.memoryUsage();
-    states.setState(`${id}.memRss`, { val: Math.round(mem.rss / 10485.76 /* 1MB / 100 */) / 100, ack: true, from: id });
-    states.setState(`${id}.memHeapTotal`, {
-        val: Math.round(mem.heapTotal / 10485.76 /* 1MB / 100 */) / 100,
-        ack: true,
-        from: id
-    });
-    states.setState(id + '.memHeapUsed', {
-        val: Math.round(mem.heapUsed / 10485.76 /* 1MB / 100 */) / 100,
-        ack: true,
-        from: id
-    });
+    try {
+        const mem = process.memoryUsage();
+        states.setState(`${id}.memRss`, {
```

Comment:
Has setState thrown then it will still throw async without await

Reply:
No the "process.memoryUsage" has thrown